### DISCUSSION TITLE
How would you achieve this design without "AndExpand" ?

### DISCUSSION THREAD

**Discussion by softlion** [dotnet/maui#9087] [General] [answered]
Hello,  
This design can not be switched from a StackLayout to a Grid, as a Grid has columns with the same width for every rows.   
And in this design, each row has different width so both the text and the control can fit in them.

On the following pic:
- 1st line: the switch has HorizontalOptions="End". No effect, it stays next to the 1st control.
- 2nd line: using EndAndExpand (deprecated), but that's what i want to achieve.
- other lines: same as 1st line.

![image](https://user-images.githubusercontent.com/190756/181903104-1e0aae54-a3e5-472e-a389-63a9094b776d.png)

So what's the alternative when AndExpand is removed ?

**Comment by sjordanGSS** [comment]
Have you tried FlexLayout? You could have one per row with FlexLayout.Grow="1" applied to the text and FlexLayout.Shrink="0" to the switch/button. That should give you something analogous to a StackLayout with AndExpand applied.

**Comment by hartez** [comment, marked as answer]
@softlion Can you post your layout (XAML or C#) that you are currently using to achieve your desired layout with "AndExpand"?
